perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/FacilitiesManagerBlImpl.java

```diff
@@ -436,12 +437,23 @@ public void deleteFacility(PerunSession sess, Facility facility, Boolean force)
 	}
 
 	@Override
-	public Facility updateFacility(PerunSession sess, Facility facility) throws FacilityExistsException {
+	public Facility updateFacility(PerunSession sess, Facility facility) throws FacilityExistsException, ConsentHubExistsException {
 		//check facility name, it can contain only a-zA-Z.0-9_-
 		if (!facility.getName().matches("^[ a-zA-Z.0-9_-]+$")) {
 			throw new IllegalArgumentException("Wrong facility name, facility name can contain only a-Z0-9.-_ and space characters");
 		}
 
+		// if renaming the facility, then rename also the corresponding consent hub
+		try {
+			if (!facility.getName().equals(getFacilitiesManagerImpl().getFacilityById(sess, facility.getId()).getName())) {
```

Comment:
maybe we should first check that the consent hub name matches the original facility name, because once we start using the joint consent hubs for more facilities they could have some special consenthub name set and this would rewrite it. what do you think @sarkapalkovicova @cuadradek

Reply:
yeah, that's a good point